### ISSUE TITLE
Counsel wipes input (counsel-rg / counsel-fzf/ async counsel commands?)

### ISSUE THREAD

**Issue by jojojames** [abo-abo/swiper#1772]
```
(setq package-enable-at-startup nil
      package-archives
      '(("melpa" . "http://melpa.milkbox.net/packages/")
        ("gnu" . "http://elpa.gnu.org/packages/")))

(require 'package)
(package-initialize)
(package-refresh-contents)
(package-install 'swiper)
(package-install 'ivy)
(package-install 'counsel)

M-x counsel-rg

Click window (e.g. *scratch* buffer) -> counsel-rg adds output to that window
```

This is a manual repro of output being added to another window other than the main ivy window. I've seen this happen even when I don't manually select a window sometimes (not all the times) with counsel-rg or counsel-fzf.

**Comment by basil-conto** [comment]
I'm sorry, but I don't think the description is clear enough.

> `counsel-rg` adds output to that window

What output? Have you typed in a search string which resulted in search results, or are you referring to e.g. `counsel-rg`'s `3 chars more` prompt?

> I've seen this happen even when I don't manually select a window sometimes (not all the times) with `counsel-rg` or `counsel-fzf.

Again, the precise set of steps and outcome are not clear. Can you please provide more details and/or screenshots/screencast?

For instance, here's what I did to try to reproduce this:

1. `make plain`.
2. <kbd>M-x</kbd>`counsel-fzf`<kbd>RET</kbd>
   * List of 26 files under repo is displayed for completion, as expected
3. Click in `*scratch*` buffer
   * Minibuffer completion is unchanged, no text appears in `*scratch*`
4. <kbd>C-x</kbd><kbd>o</kbd>`ivy`
   * Completion is narrowed to files which include `ivy` in their name
5. Click in `*scratch*` buffer
   * Minibuffer completion is unchanged, no text appears in `*scratch*`

Here's a more programmatic example:

1. `make plain`
2. ```el
   (progn (setq unread-command-events
                (listify-key-sequence "\C-xo"))
          (call-interactively #'counsel-fzf))
   ```
3. <kbd>C-x</kbd><kbd>C-e</kbd>
   * Minibuffer completion starts, but no candidates are displayed yet (this could be considered a separate bug)
   * `*scratch*` is selected, but no additional text has been printed to it
4. <kbd>C-x</kbd><kbd>o</kbd>
   * Minibuffer is selected and candidates are finally displayed

Replacing `(listify-key-sequence "\C-xo")` with something more elaborate like `(listify-key-sequence "ivy\C-xo\C-xo")` still works as expected.

Similarly, `counsel-rg` seems to behave as expected:

1. `make plain`
2. <kbd>C-c</kbd><kbd>k</kbd>
   * Minibuffer completion starts and `3 chars more` is displayed
3. Click in `*scratch*`
   * Minibuffer completion is unchanged, no text appears in `*scratch*`
4. <kbd>C-x</kbd><kbd>o</kbd>`ivy`
   * Completion is narrowed occurences of `ivy`
5. Click in `*scratch*` buffer
   * Minibuffer completion is unchanged, no text appears in `*scratch*`

And the same can be seen with e.g.
```el
(progn (setq unread-command-events
             (listify-key-sequence "ivy\C-xo\C-xo"))
       (call-interactively #'counsel-rg))
```

**Comment by jojojames** [comment]
> What output? Have you typed in a search string which resulted in search results, or are you referring to e.g. counsel-rg's 3 chars more prompt?

Yes 'asdf' will do. Of course, the search string will be dependent on whatever your filesystem may look like.

> Again, the precise set of steps and outcome are not clear. 

It happens 'every so often' so I'm not going to be able to provide a list of steps for that case but I'd expect but cases to be symptomatic of the same issue.

> Can you please provide more details and/or screenshots/screencast?

Yeah.

https://www.dropbox.com/s/zdojsha800quiw2/counsel_bug.mov?dl=0

**Comment by basil-conto** [comment]
>> Can you please provide more details and/or screenshots/screencast?

> Yeah.

> https://www.dropbox.com/s/zdojsha800quiw2/counsel_bug.mov?dl=0

Wow, I didn't expect such bad behaviour. :) This has never happened to me, and I'm worried it might be hard to debug. It may even be OS-specific. Any programmatic reproduction recipe you can come up with will prove very helpful.

Does the issue occur in the absence of other packages except Ivy/Swiper/Counsel? I.e. can you reproduce the issue from `emacs -Q` followed by manually loading only Ivy/Swiper/Counsel? If not, there may be some other package you have installed that is interfering.

**Comment by basil-conto** [comment]
What is your <kbd>M-x</kbd>`emacs-version`<kbd>RET</kbd>?

**Comment by basil-conto** [comment]
> The video/recipe is starting from `emacs -Q`.

But you call `package-initialize`, no? This will evaluate the autoloads of all installed third-party packages, potentially (even if only by a small chance) interfering with the recipe.

**Comment by jojojames** [comment]
That's a good point. I removed all my packages from my elpa directory and the problem persists.

**Comment by ChoppinBlockParty** [comment]
Ican confirm that i experienced this issue with fzf. I start counsel-fzf in large directory tree, type something than quit minibuffer while fzf was still working. I get the prompt updates added to current buffer. That does not happen often, so it is hard to reproduce.

**Comment by abo-abo** [comment]
Please test if the issue is still there.

**Comment by mookid** [comment]
I confirm it is still there.

**Comment by jojojames** [comment]
The repro steps here don't repro anymore for me but I trust @mookid 's judgement in #1857.

**Comment by mookid** [comment]
I suggest closing this for now.

**Comment by abo-abo** [comment]
@mookid OK, closing.

Thanks, all.
